schema/json_test.go

```diff
@@ -43,6 +43,8 @@ func TestJSONSet(t *testing.T) {
 		{source: map[string]Foo{}, result: JSON{Bytes: []byte(`{}`), Status: Present}},
 
 		{source: nil, result: JSON{Status: Null}},
+
+		{source: map[string]any{"test1": "a&b", "test2": "😀"}, result: JSON{Bytes: []byte(`{"test1": "a&b", "test2": "😀"}`), Status: Present}},
```

Comment:
😀

Reply:
I assume this test failed before the fix? 🤔

Reply:
Yes, it failed due to the `&` character that was escaped. The emoji is somehow redundant for the purpose of the failure but I wanted to verify it doesn't get escaped too

Reply:
Kindly add the test from https://github.com/cloudquery/cloudquery/pull/7815 as well

Reply:
Can you please be specific on which test to add? I think with this fix:
1. Unescaping `{"BillTo":"P\u0026T"}` is not needed at the destination, as the cause was the redundant escaping fixed by this PR
2. `{"BillTo":"P&T"}` should be covered by the test here
3. Not sure why we need to unquote, is there a bug report I can use to reproduce?

Reply:
The unquote one. Basically, it was produced by parsing an AWS output, as the original field was a string representing the JSON contents, and it's returned from the API/SDK escaped/quoted